bazel/packaging/BUILD

```diff
@@ -0,0 +1,34 @@
+load("@rules_pkg//pkg:mappings.bzl", "pkg_files", "strip_prefix")
+load("@rules_pkg//pkg:tar.bzl", "pkg_tar")
+
+pkg_files(
+    name = "redpanda_binary_with_runfiles",
+    srcs = [
+        "//src/v/redpanda",
+    ],
+    # Including runfiles has the effect of copying all of the shared libraries needed to run the binary.
+    # Although they always retain the directory structure that is found in bazel build output.
+    include_runfiles = True,
+)
+
+pkg_files(
+    name = "redpanda_binary_flattened",
+    srcs = [
+        ":redpanda_binary_with_runfiles",
+    ],
+    prefix = "opt/redpanda/libexec",
+    # This flattens the files so that everything is directly in the `libexec` folder.
+    strip_prefix = strip_prefix.files_only(),
```

Comment:
interesting. what's the motivation for having flattened and not flattened pkg_files?

Reply:
I need to look again for the exact structure of our current package, but we will need to set LD_LIBRARY_PATH to override where shared libraries are loaded from, and I only want to set that to be a single directory (I don't even know if you can do multiple).

Maybe I would wait to make more progress on packaging before merging? I had wanted to commit the progress I had made and it felt like a decent size to get the binary and bazel built so into a tarball.

Reply:
I added more comments, is this more clear now?

Reply:
I made a bit more progress so I think this is in a good state to check in and get plat eng eyes on it